generator/src/main/java/com/scottlogic/deg/generator/decisiontree/DecisionTreeOptimiser.java

```diff
@@ -51,7 +51,8 @@ private boolean noChangeInDecisionCount(ConstraintNode rootNode, ConstraintNode
     }
 
     private ConstraintNode optimiseDecisions(ConstraintNode rootNode){
-        AtomicConstraint mostProlificAtomicConstraint = getMostProlificAtomicConstraint(rootNode.getDecisions());
+        List<DecisionNode> decisions = rootNode.getDecisions().stream().distinct().collect(Collectors.toList());
+        AtomicConstraint mostProlificAtomicConstraint = getMostProlificAtomicConstraint(decisions);
         if (mostProlificAtomicConstraint == null){
```

Comment:
I'm not sure that we actually want to make this change, because getting a single AtomicConstraint shouldn't be affected if the decisions are duplicated.

Reply:
The intent here is if during the optimisation process some of the decisions have been reduced into the same choice then we can remove them from the decision tree. This might not be the most appropriate place for this check. Is there a better place to do this sort of check?

Reply:
Agreed. I've removed this optimisation.